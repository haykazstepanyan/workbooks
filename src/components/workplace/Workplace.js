import PropTypes from "prop-types";
import { useCallback } from "react";
import { useDispatch } from "react-redux";
import {
	deleteWorkplace,
	resetDeleteStatus,
} from "../../redux/workplace/actionCreators";
import Accordion from "../accordion";
import Table from "../table";
import styles from "./workplaces.module.scss";

function Workplace({ workplaceID, workbookID, workplace, canDelete }) {
	const { company, ...workplaceRest } = workplace;

	const dispatch = useDispatch();

	const tableData = {
		...workplaceRest,
		startDate: new Date(workplaceRest.startDate).toDateString().slice(4),
	};

	if (workplaceRest.endDate) {
		tableData.endDate = new Date(workplaceRest.endDate).toDateString().slice(4);
	}

	const handleDeleteWorkplace = useCallback(() => {
		dispatch(deleteWorkplace(workplaceID, workbookID));
		setTimeout(() => {
			dispatch(resetDeleteStatus());
		}, 5000);
	}, [dispatch, workplaceID, workbookID]);

	return (
		<div className={styles.workplace}>
			<Accordion
				rowName={workplace.company}
				fullWidth
				canDelete={canDelete}
				handleDelete={handleDeleteWorkplace}
			>
				<Table tableData={tableData} />
			</Accordion>
		</div>
	);
}

Workplace.propTypes = {
	workplaceID: PropTypes.string,
	workbookID: PropTypes.string,
	workplace: PropTypes.objectOf(PropTypes.string),
	canDelete: PropTypes.bool,
};

export default Workplace;
