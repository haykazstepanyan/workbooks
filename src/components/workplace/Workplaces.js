import PropTypes from "prop-types";
import classNames from "classnames";
import AddNewWorkplace from "./AddNewWorkplace";
import Workplace from "./Workplace";
import styles from "./workplaces.module.scss";

function Workplaces({ workbookID, workplaces, user, workbookUserID }) {
	let canCreateNewWorkplace = true;
	if (user.type === "user") {
		canCreateNewWorkplace = user.id === workbookUserID;
	}

	const divClass = classNames({
		[styles.workplaces]: Boolean(workplaces),
	});

	return (
		<>
			<div className={divClass} data-testid="workplaces">
				{canCreateNewWorkplace && (
					<AddNewWorkplace id={workbookID} workplaces={workplaces} />
				)}
				{workplaces &&
					Object.entries(workplaces).map(([key, value]) => (
						<Workplace
							userID={user.id}
							workbookID={workbookID}
							workplaceID={key}
							workplace={value}
							key={key}
							canDelete={canCreateNewWorkplace}
						/>
					))}
			</div>
		</>
	);
}

Workplaces.propTypes = {
	workbookID: PropTypes.string,
	workplaces: PropTypes.objectOf(
		PropTypes.shape({
			country: PropTypes.string,
			company: PropTypes.string,
			startDate: PropTypes.string,
			endDate: PropTypes.string,
		})
	),
	user: PropTypes.objectOf(PropTypes.string),
	workbookUserID: PropTypes.string,
};

export default Workplaces;
