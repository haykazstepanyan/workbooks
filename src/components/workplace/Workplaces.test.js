import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Workplaces from "./";
import Workplace from "./Workplace";

describe("workplace folder components", () => {
	const mockStore = configureStore();
	const store = mockStore({ workplace: {} });

	describe("Workplaces component", () => {
		it("should match component snapshot", () => {
			const { getByTestId } = render(
				<Provider store={store}>
					<Workplaces user={{}} />
				</Provider>
			);
			const workplaces = getByTestId("workplaces");
			expect(workplaces).toMatchSnapshot();
		});
	});

	describe("Workplace component", () => {
		it("should match Workplace component snapshot", () => {
			const { getByTestId } = render(
				<Provider store={store}>
					<Workplace />
				</Provider>
			);
			const workplace = getByTestId("workplace");
			expect(workplace).toMatchSnapshot();
		});
	});
});
