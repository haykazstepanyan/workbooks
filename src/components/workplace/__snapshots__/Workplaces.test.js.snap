// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workplace folder components Workplace component should match Workplace component snapshot 1`] = `
<div
  class="workplace"
  data-testid="workplace"
>
  <div
    class="MuiPaper-root MuiAccordion-root accordion accordion__fullWidth MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root accordion__content"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root accordion__details"
            >
              <ul
                class="table"
                data-testid="table"
              >
                <li>
                  <span>
                    startDate
                    :
                  </span>
                  <span>
                    lid Date
                  </span>
                </li>
                <li>
                  <span>
                    endDate
                    :
                  </span>
                  <span>
                    lid Date
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`workplace folder components Workplaces component should match component snapshot 1`] = `
<div
  class=""
  data-testid="workplaces"
>
  <div
    class="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root accordion__content"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          + Add new workplace
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root accordion__details"
            >
              <form
                class="forms"
              >
                <div
                  class="inputs"
                  data-testid="wrapper-box"
                >
                  <input
                    class="inputs__field"
                    data-testid="input"
                    id="company"
                    type="text"
                    value=""
                  />
                  <label
                    class="inputs__labels"
                    data-testid="label"
                    for="company"
                  >
                    Company
                  </label>
                </div>
                <div
                  class="inputs"
                  data-testid="wrapper-box"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class="inputs__field"
                        id="startDate"
                        type="text"
                        value="01/07/2021"
                      />
                    </div>
                  </div>
                  <label
                    class="inputs__labels"
                    data-testid="label"
                    for="startDate"
                  >
                    Start date
                  </label>
                </div>
                <div
                  class="inputs inputs_zeroMargin"
                  data-testid="wrapper-box"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class="inputs__field"
                        id="endDate"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <label
                    class="inputs__labels"
                    data-testid="label"
                    for="endDate"
                  >
                    End date
                  </label>
                </div>
                <div
                  class="inputs inputs_directionRow"
                  data-testid="wrapper-box"
                >
                  <input
                    class="inputs__field"
                    data-testid="input"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="inputs__labels"
                    data-testid="label"
                  >
                    Working till now
                  </label>
                </div>
                <div>
                  <div
                    class="ReactFlagsSelect-module_flagsSelect__2pfa2 workplaces__flagSelect"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="rfs_btn"
                      class="ReactFlagsSelect-module_selectBtn__19wW7"
                      id="rfs_btn"
                      style="font-size: 16px;"
                      type="button"
                    >
                      <span
                        class="ReactFlagsSelect-module_selectValue__152eS"
                      >
                        Select a country
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="buttons_rightAlign"
                >
                  <button
                    class="buttons"
                    data-testid="button"
                    disabled=""
                    type="submit"
                  >
                    Add
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
